checksec.cpp

```diff
@@ -161,6 +162,65 @@ Checksec::operator json() const {
     };
 }
 
+const std::string Checksec::detailedReport() const {
```

Comment:
I don't think we currently want this functionality: it'll be a little unwieldy to maintain (since we don't currently have a nice way to just enumerate all of our checks), and its existence on the main `Checksec` class is another place where we fail to isolate with CLI from the C++ public API.

I'm happy to merge the rest of this PR, though.